### PR TITLE
DevOps Principles - Draft Ready for Review.

### PR DESCRIPTION
**Pull request by ghost**
🚀 Initial draft of the `DevOps Principles` page. Requires a review please. 🔍

### CODE REVIEW

**Review by chelnak — APPROVED** [review]
Nice @dalemorson.

I noticed one or two Apprenticeship service where it should be Apprenticeship Service (capital S) but that doesn’t impact approval.

**Comment by ghost** [comment]
> Nice @dalemorson.
> 
> I noticed one or two Apprenticeship service where it should be Apprenticeship Service (capital S) but that doesn’t impact approval.

Thanks for reviewing @chelnak I've only just seen this! Will check out those few grammar issues cheers dude!!

File: principles/devops.md [hunk 1]
```diff
@@ -5,5 +5,52 @@ title:  "DevOps Principles"
...
+
+## Automation
+
+This is the most vital factor of the DevOps culture. The purpose of automation is to build a system such that it eliminates various human errors and can fix errors and repeating the successful process again without any sort of intervention or delays. Automation greatly improves the workflow and productivity of the service. 
+
+- **Infrastructure as Code (IaC).** IaC is an infrastructure management approach that makes continuous delivery and DevOps possible. Without IaC, our developers and engineers would have to treat each target environment individually, which becomes a tedious task as we have different environments for development, testing, and production use. We include the Azure Resource Manager (ARM) templates and Azure DevOps build definitions along side each an applications repository in GitHub, this is to match the development lifecycle of an application. As application requirements change, so can the build and release along side it.
```
Comment (DannyGB) [inline principles/devops.md]: bit of grammar: `...along side each an applications repository...`

File: principles/devops.md [hunk 2]
```diff
@@ -5,5 +5,52 @@ title:  "DevOps Principles"
...
+
+- **Show and tell.** These are regular meetings that allow teams to celebrate their work and talk about what they’ve learned. It's a chance to bring teams together, share success and improve on collaboration. Progress is highlighted along with question and feedback sessions. Transparency is encouraged to glimpse 'behind the curtain' of the products in development and to highlight where dependencies may exist between teams. A great show and tell session is one that is performed on a regular basis to ensure continuous feedback.
+
+## Automation
+
+This is the most vital factor of the DevOps culture. The purpose of automation is to build a system such that it eliminates various human errors and can fix errors and repeating the successful process again without any sort of intervention or delays. Automation greatly improves the workflow and productivity of the service.
```
Comment (DannyGB) [inline principles/devops.md]: bit of an awkward sentence: `...human errors and can fix errors and repeating the successful process again...`

**Review by DannyGB — APPROVED** [review]

File: principles/devops.md (hunk 2 above)
Comment (ghost) [reply principles/devops.md]: Updated 👍

File: principles/devops.md (hunk 1 above)
Comment (ghost) [reply principles/devops.md]: Sorted 👍

File: principles/devops.md [hunk 3]
```diff
@@ -5,5 +5,52 @@ title:  "DevOps Principles"
...
 # DevOps
 
-Placeholder
+## Introduction
 
+**DevOps is a service wide way of working, we are all in this together.** It is not the tools, nor is it a single team. It is famously summarised in a single sentance by Donovan Brown (Principal DevOps Manager at Microsoft):
+
+> DevOps is the union of people, process, and products to enable continuous delivery of value to our end users.
+
+We follow this ethos to ensure delivery of value to all of the Digital Apprenticeship Service users, internal and external. It is a combination of the passion, hard work and process that enables DevOps culture within the service. The continuous set of processes that we follow forms the methodology for our Agile planning and continuous delivery, for example aligned sprints across all teams.
```
Comment (CloudPlatformer) [inline principles/devops.md]:
```suggestion
We follow this ethos to ensure delivery of value to all of the Apprenticeship Service users, internal and external. It is a combination of the passion, hard work and process that enables DevOps culture within the service. The continuous set of processes that we follow forms the methodology for our Agile planning and continuous delivery, for example aligned sprints across all teams.
```

File: principles/devops.md [hunk 4]
```diff
@@ -5,5 +5,52 @@ title:  "DevOps Principles"
...
+
+Adhereing to DevOps provides the service with the following benefits:
```
Comment (CloudPlatformer) [inline principles/devops.md]:
```suggestion
Adhering to DevOps provides the service with the following benefits:
```

File: principles/devops.md [hunk 5]
```diff
@@ -5,5 +5,52 @@ title:  "DevOps Principles"
...
+
+## Automation
+
+This is the most vital factor of the DevOps culture. The purpose of automation is to build a system that eliminates various human errors and provide a successful repeating process with no manual intervention required. Automation greatly improves the workflow and productivity of the service. 
+
+- **Infrastructure as Code (IaC).** IaC is an infrastructure management approach that makes continuous delivery and DevOps possible. Without IaC, our developers and engineers would have to treat each target environment individually, which becomes a tedious task as we have different environments for development, testing, and production use. We include the Azure Resource Manager (ARM) templates and Azure DevOps build definitions along side an applications codebase in the GitHub repository, this is to match the development lifecycle of an application. As application requirements change, so can the build and release along side it.
```
Comment (CloudPlatformer) [inline principles/devops.md]:
```suggestion
- **Infrastructure as Code (IaC).** IaC is an infrastructure management approach that makes continuous delivery and DevOps possible. Without IaC, our developers and engineers would have to treat each target environment individually, which becomes a tedious task as we have different environments for development, testing, and production use. We include the Azure Resource Manager (ARM) templates and Azure DevOps build definitions alongside an application's codebase in the GitHub repository, this is to match the development lifecycle of an application. As application requirements change, so can the build and release alongside it. 
```

File: principles/devops.md [hunk 6]
```diff
@@ -5,5 +5,52 @@ title:  "DevOps Principles"
...
+
+- **Infrastructure as Code (IaC).** IaC is an infrastructure management approach that makes continuous delivery and DevOps possible. Without IaC, our developers and engineers would have to treat each target environment individually, which becomes a tedious task as we have different environments for development, testing, and production use. We include the Azure Resource Manager (ARM) templates and Azure DevOps build definitions along side an applications codebase in the GitHub repository, this is to match the development lifecycle of an application. As application requirements change, so can the build and release along side it. 
+
+- **Cloud infrastructure.** We are a cloud first service as it provides the flexibility, toolsets, and scalability required. All our architectures rely on PaaS solutions. Through IaC and continuous integration and continuous deployment (CI/CD) we can quickly deploy infrastructure and ensure there is no environment or configuration drift. Builds can be deployed to each environment providing consistant dev/test and production environments. It enables our teams to test applications in production-like environments early in the development cycle.
```
Comment (CloudPlatformer) [inline principles/devops.md]:
```suggestion
- **Cloud infrastructure.** We are a cloud first service as it provides the flexibility, toolsets, and scalability required. All our architectures rely on PaaS solutions. Through IaC and continuous integration and continuous deployment (CI/CD) we can quickly deploy infrastructure and ensure there is no environment or configuration drift. Builds can be deployed to each environment providing consistent dev/test and production environments. It enables our teams to test applications in production-like environments early in the development cycle.
```

**Review by CloudPlatformer — APPROVED** [review]
Great write-up, I've commented some minor edits.

**Comment by ghost** [comment]
> Great write-up, I've commented some minor edits.

Cheers @jack-education for reviewing, I've committed your suggestions 👍

**Pull request merged**